docs/observability/cloud-trace.md

```diff
@@ -0,0 +1,227 @@
+app: FastAPI = get_fast_api_app(
+    agents_dir=AGENT_DIR,
+    web=True,
+    trace_to_cloud=True,
+)
+
+app.title = "weather-agent"
+app.description = "API for interacting with the Agent weather-agent"
+
+
+# Main execution
+if __name__ == "__main__":
+    import uvicorn
+
+    uvicorn.run(app, host="0.0.0.0", port=8080)
+```
+
+
+### From Customized Agent Runner
+
+If you want to fully customize your ADK agent runtime, you also can enable cloud tracing by using `CloudTraceSpanExporter` module from Opentelemetry.
+
+```python
+# agent_runner.py
+
+from google.adk.runners import Runner
+from google.adk.sessions import InMemorySessionService
+from weather_agent.agent import root_agent as weather_agent
+from google.genai.types import Content, Part
+from opentelemetry import trace
+from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+from opentelemetry.sdk.trace import export
+from opentelemetry.sdk.trace import TracerProvider
+
+APP_NAME = "weather_agent"
+USER_ID = "u_123"
+SESSION_ID = "s_123"
+
+provider = TracerProvider()
+processor = export.BatchSpanProcessor(
+    CloudTraceSpanExporter(project_id="{your-project-id}")
+)
+provider.add_span_processor(processor)
+trace.set_tracer_provider(provider)
+
+session_service = InMemorySessionService()
+runner = Runner(agent=weather_agent, app_name=APP_NAME, session_service=session_service)
+
+
+async def main():
+    session = await session_service.get_session(
+        app_name=APP_NAME, user_id=USER_ID, session_id=SESSION_ID
+    )
+    if session is None:
+        session = await session_service.create_session(
+            app_name=APP_NAME, user_id=USER_ID, session_id=SESSION_ID
+        )
+
+    user_content = Content(
+        role="user", parts=[Part(text="hello, what's weather in paris?")]
+    )
+
+    final_response_content = "No response"
+    async for event in runner.run_async(
+        user_id=USER_ID, session_id=SESSION_ID, new_message=user_content
+    ):
+        if event.is_final_response() and event.content and event.content.parts:
+            final_response_content = event.content.parts[0].text
+
+    print(final_response_content)
+
+
+if __name__ == "__main__":
+    import asyncio
+
+    asyncio.run(main())
+```
+
+## Inspect Cloud Traces
+
+After the setup is complete, whenever you interact with the agent it will automatically send trace data to Cloud Trace. You can inspect the traces by going to [console.cloud.google.com](https://console.cloud.google.com) and visit the Trace Explorer on the configured Google Cloud Project
+
+![cloud-trace](../assets/cloud-trace1.png)
+
+And then you will see all available traces produced by ADK agent which configured in several span names such as `invocation` , `agent_run` . `call_llm` and `execute_tool`
+
+![cloud-trace](../assets/cloud-trace2.png)
+
+If you click on one of the traces, you will see the waterfall view of the detailed process, kinda similar of what we can see in the web development UI
```

Comment:
"kinda similar of what we can see in the web development UI"
->
"similar to what we see in the web development UI with adk web."